dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java

```diff
@@ -47,10 +47,33 @@ public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilde
      *            the record defintion for the dataset
      * @param recBuilder
      *            the IRecordBuilder used to construct records
+     * @param connectTimeout 
+     *            the connectiont timeout in seconds
      * @param projectedRD
      *            the requested record format
      */
     public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilder recBuilder, int connectTimeout) throws Exception
+    {
+        this(dp,originalRD,recBuilder,connectTimeout,-1);
+    }
+
+    /**
+     * A remote file reader that reads the part identified by the HpccPart object using the record definition provided.
+     * 
+     * @param dp
+     *            the part of the file, name and location
+     * @param originalRD
+     *            the record defintion for the dataset
+     * @param recBuilder
+     *            the IRecordBuilder used to construct records
+     * @param projectedRD
+     *            the requested record format
+     * @param connectTimeout 
+     *            the connectiont timeout in seconds
+     * @param limit 
+     *            the record limit for reading. -1 specifies no limit
```

Comment:
If we can, let's be as descriptive here with the behavior of the limit... ie. first n recs from 1st file part, etc.